File: javascript/collatz-conjecture/collatz-conjecture.js
export const steps = (positiveNumber) => {
	if (positiveNumber <= 0) { throw 'Only positive numbers are allowed'; }

	return positiveNumber === 1 ? 0 : 1 + (positiveNumber % 2 === 0 ? steps(positiveNumber / 2) : steps(3 * positiveNumber + 1));
};

File: python/collatz-conjecture/collatz_conjecture.py
def steps(number: int) -> int:
    if number < 1:
        raise ValueError("Only positive integers are allowed")
    
    steps = 0
   
    while number != 1:
        if number % 2 == 1:
            number = 3 * number + 1           
        else:
            number //=  2

        steps += 1
    
    return steps


